x/fswap/keeper/genesis_test.go

```diff
@@ -45,3 +51,78 @@ func (s *KeeperTestSuite) TestInitAndExportGenesis() {
 	s.Require().Equal(testGenesis.GetSwapStats(), exportGenesis.GetSwapStats())
 	s.Require().Equal(testGenesis.GetSwappeds(), exportGenesis.GetSwappeds())
 }
+
+func TestInitGenesis(t *testing.T) {
+	generateState := func() *types.GenesisState {
+		testSwapRate, _ := sdk.NewDecFromStr("1234567890")
+		return &types.GenesisState{
+			Swaps: []types.Swap{
+				{
+					FromDenom:           "aaa",
+					ToDenom:             "bbb",
+					AmountCapForToDenom: sdk.NewInt(1234567890000),
+					SwapRate:            testSwapRate,
+				},
+			},
+			SwapStats: types.SwapStats{
+				SwapCount: 1,
+			},
+			Swappeds: []types.Swapped{
+				{
+					FromCoinAmount: sdk.Coin{
+						Denom:  "aaa",
+						Amount: sdk.ZeroInt(),
+					},
+					ToCoinAmount: sdk.Coin{
+						Denom:  "bbb",
+						Amount: sdk.ZeroInt(),
+					},
+				},
+			},
+		}
+	}
```

Comment:
Could this function as a private function of `genesis_test` and also used in `TestInitAndExportGenesis()`